src/Shared/HandyControl_Shared/Controls/Window/Window.cs

```diff
@@ -361,23 +374,69 @@ private IntPtr HwndSourceHook(IntPtr hwnd, int msg, IntPtr wparam, IntPtr lparam
                     WmGetMinMaxInfo(hwnd, lparam);
                     Padding = WindowState == WindowState.Maximized ? WindowHelper.WindowMaximizedPadding : _commonPadding;
                     break;
+                #region SnapLayout
                 case InteropValues.WM_NCHITTEST:
-                    // for fixing #886
-                    // https://developercommunity.visualstudio.com/t/overflow-exception-in-windowchrome/167357
                     try
                     {
-                        _ = lparam.ToInt32();
+                        int x = lparam.ToInt32() & 0xffff;
+                        if (IsWin11SnaplayoutSupported())
+                        {
+                            int y = lparam.ToInt32() >> 16;
+                            var DPI_SCALE = DpiHelper.LogicalToDeviceUnitsScalingFactorX;
+                            var _button = WindowState == WindowState.Maximized ? _ButtonRestore : _ButtonMax;
+                            var rect = new Rect(_button.PointToScreen(
```

Comment:
How about cache the Rect result? Calculation costs some resources and WM_NCHITTEST is called very frequently.

Reply:
@lindexi I think caching does not work properly,  I used the following code
```
if (rect.IsEmpty)
{
    rect = new Rect(_button.PointToScreen(
    new Point()),
    new Size(_button.Width * DPI_SCALE, _button.Height * DPI_SCALE));
}
```
But by resizing/moving window, the coordinates change and cached rect does not work and we need to call `rect = new Rect(_button.PointToScreen(` again.
Maybe I'm doing the caching operation wrong😕